pkg/operator/csi/csiconfigobservercontroller/csi_config_observer_controller.go

```diff
@@ -66,16 +84,23 @@ func NewCSIConfigObserverController(
 			operatorClient,
 			eventRecorder.WithComponentSuffix("csi-config-observer-controller-"+strings.ToLower(name)),
 			Listers{
-				ProxyLister_: configinformers.Config().V1().Proxies().Lister(),
+				APIServerLister_: configinformers.Config().V1().APIServers().Lister(),
+				ProxyLister_:     configinformers.Config().V1().Proxies().Lister(),
 				PreRunCachesSynced: append([]cache.InformerSynced{},
 					operatorClient.Informer().HasSynced,
 					configinformers.Config().V1().Proxies().Informer().HasSynced,
+					configinformers.Config().V1().APIServers().Informer().HasSynced,
 				),
 			},
 			informers,
 			proxy.NewProxyObserveFunc(ProxyConfigPath()),
+			observeTLSSecurityProfile,
 		),
 	}
 
 	return c
 }
+
+func observeTLSSecurityProfile(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
+	return libgoapiserver.ObserveTLSSecurityProfileWithPaths(genericListers, recorder, existingConfig, MinTLSVersionPath(), CipherSuitesPath())
```

Comment:
An alternative would be to extract the expected configuration here and pass extracted/expected cfg to the existing function. Although this could be more error prone because you would have to remember to prune the cfg and handle some errors. WDYT ?

Reply:
@p0lyn0mial Can you please elaborate? To what *existing function* should we pass the extracted configuration?

I thought it's point of an observer to *return* the extracted config, so the common observer can put it into `spec.observedConfig`.

Reply:
Yeah, I'd like Fabio's version, IMO it's more explicit.
/hold cancel
(he's on PTO)